docs/source/how-to-upstream.mdx

```diff
@@ -24,6 +26,34 @@ Whenever you want to upload files to the Hub, you need to log in to your Hugging
 
    [`notebook_login`] will launch a widget in your notebook from which you can enter your Hugging Face credentials.
 
+## Upload a file
+
+Use the [`upload_file`] function to upload a file to a repository. Let the function know where the file you want to upload is, where you want to upload the file to in the repository, and the name of the repository you want to add the file to.
```

Comment:
Would be great to clarify here whether the repo_id needs to already exist on the hub, or whether a repo will be created automatically in case it doesn't exist already.

Cause personally I don't know actually :D

Reply:
For creating a repository they should likely use `create_repo`. `Repository` is git-based and would clone the repo locally. `create_repo` + `upload_file` take care of it.

Reply:
Yes, `Repository` shouldn't be used to create the repository. It can create it, but this feature is likely to disappear in favor of the unambiguous `create_repo` instead (see discussions in https://github.com/huggingface/huggingface_hub/issues/672)